### DISCUSSION TITLE
battery % or charging bugs??

### DISCUSSION THREAD

**Discussion by frozennutzz** [seerge/g-helper#3486] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/7aefe37e-e232-4fd0-9909-3866cb31ac68)
can someone tell me if its a G helper bugs or the 24H2 bugs or my laptop fault??!!

**Comment by seerge** [comment]
@frozennutzz hello,

Charge percentage is taken from windows API (you can check it in app log under `%appdata%\GHelper\log.txt`). It could be that windows widget simply rounds 98% to a 100%.

Try to check charge with other independent source, like HWinfo app.

**Reply by frozennutzz** [reply]
so does that means the laptop is not charging 100%
sorry for these silly questions bro im absolutely new into this.

**Reply by seerge** [reply, marked as answer]
@frozennutzz it is charging to the max. Don't forget that windows estimate of a battery health and a battery charge is an **approximate** value. 

System calibrates (measures) things on full discharge-charge cycles periodically, so if you will discharge laptop to a low values and then charge again - numbers that you see may become a bit more accurate.

**Reply by frozennutzz** [reply]
![image](https://github.com/user-attachments/assets/46974b70-ed35-4a24-b2df-3d7772ce77e5)
thank u bro the battery % issue is solved by now.
but im i noticed that the CPU temp is missing..!!
i thought it might be a glitch or something and restarted the laptop 2 to 3 times but still the same issue...

**Reply by seerge** [reply]
@frozennutzz CPU utilization has never been there :) I don't want to overload UI with metrics that majority don't need

**Reply by frozennutzz** [reply]
sorry for my mistake bro😅what i mean is CPU TEMPERATURE...
